crates/ruff_linter/src/rules/pycodestyle/rules/logical_lines/blank_lines.rs

```diff
@@ -0,0 +1,620 @@
+///
+/// ## References
+/// - [PEP 8](https://peps.python.org/pep-0008/#blank-lines)
+/// - [Flake 8 rule](https://www.flake8rules.com/rules/E304.html)
+#[violation]
+pub struct BlankLineAfterDecorator;
+
+impl AlwaysFixableViolation for BlankLineAfterDecorator {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("blank lines found after function decorator")
+    }
+
+    fn fix_title(&self) -> String {
+        "Remove extraneous blank line(s)".to_string()
+    }
+}
+
+/// ## What it does
+/// Checks for missing blank lines after end of function or class.
+///
+/// ## Why is this bad?
+/// PEP 8 recommends the using blank lines as following:
+/// - Two blank lines are expected between functions and classes
+/// - One blank line is expected between methods of a class.
+///
+/// ## Example
+/// ```python
+/// class User(object):
+///     pass
+/// user = User()
+/// ```
+///
+/// Use instead:
+/// ```python
+/// class User(object):
+///     pass
+///
+///
+/// user = User()
+/// ```
+///
+/// ## References
+/// - [PEP 8](https://peps.python.org/pep-0008/#blank-lines)
+/// - [Flake 8 rule](https://www.flake8rules.com/rules/E305.html)
+#[violation]
+pub struct BlankLinesAfterFunctionOrClass(pub u32);
+
+impl AlwaysFixableViolation for BlankLinesAfterFunctionOrClass {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let BlankLinesAfterFunctionOrClass(blank_lines) = self;
+        format!("expected 2 blank lines after class or function definition, found ({blank_lines})")
+    }
+
+    fn fix_title(&self) -> String {
+        "Add missing blank line(s)".to_string()
+    }
+}
+
+/// ## What it does
+/// Checks for for 1 blank line between nested functions/classes definitions.
+///
+/// ## Why is this bad?
+/// PEP 8 recommends the using blank lines as following:
+/// - Two blank lines are expected between functions and classes
+/// - One blank line is expected between methods of a class.
+///
+/// ## Example
+/// ```python
+/// def outer():
+///     def inner():
+///         pass
+///     def inner2():
+///         pass
+/// ```
+///
+/// Use instead:
+/// ```python
+/// def outer():
+///     def inner():
+///         pass
+///
+///     def inner2():
+///         pass
+/// ```
+///
+/// ## References
+/// - [PEP 8](https://peps.python.org/pep-0008/#blank-lines)
+/// - [Flake 8 rule](https://www.flake8rules.com/rules/E306.html)
+#[violation]
+pub struct BlankLinesBeforeNestedDefinition(pub u32);
+
+impl AlwaysFixableViolation for BlankLinesBeforeNestedDefinition {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let BlankLinesBeforeNestedDefinition(blank_lines) = self;
+        format!("Expected 1 blank line before a nested definition, found {blank_lines}")
+    }
+
+    fn fix_title(&self) -> String {
+        "Add missing blank line".to_string()
+    }
+}
+
+/// Returns `true` if line is a docstring only line.
+fn is_docstring(line: Option<&LogicalLine>) -> bool {
```

Comment:
This method seems to always be called with `Some(line)`. Thus, using `Option` is unnecessary
```suggestion
fn is_docstring(line: &LogicalLine) -> bool {
```